src/main/java/org/opensearch/geospatial/search/aggregations/bucket/geogrid/GeoHexGridAggregationBuilder.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.search.aggregations.bucket.geogrid;
+
+import static org.opensearch.geospatial.search.aggregations.bucket.geogrid.GeoHexHelper.checkPrecisionRange;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.opensearch.OpenSearchParseException;
+import org.opensearch.common.geo.GeoBoundingBox;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.xcontent.ObjectParser;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.support.XContentMapValues;
+import org.opensearch.geo.search.aggregations.bucket.geogrid.GeoGridAggregationBuilder;
+import org.opensearch.geo.search.aggregations.metrics.GeoGridAggregatorSupplier;
+import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.search.aggregations.AggregationBuilder;
+import org.opensearch.search.aggregations.AggregatorFactories;
+import org.opensearch.search.aggregations.AggregatorFactory;
+import org.opensearch.search.aggregations.support.ValuesSourceAggregatorFactory;
+import org.opensearch.search.aggregations.support.ValuesSourceConfig;
+import org.opensearch.search.aggregations.support.ValuesSourceRegistry;
+
+/**
+ * Aggregation Builder for geo hex grid
+ */
+public class GeoHexGridAggregationBuilder extends GeoGridAggregationBuilder {
+    private static final int DEFAULT_MAX_NUM_CELLS = 10000;
+    private static final int DEFAULT_PRECISION = 5;
+    private static final int DEFAULT_SHARD_SIZE = -1;
+    public static final String NAME = "geohex_grid";
+    public static final ValuesSourceRegistry.RegistryKey<GeoGridAggregatorSupplier> REGISTRY_KEY = new ValuesSourceRegistry.RegistryKey<>(
+        NAME,
+        GeoGridAggregatorSupplier.class
+    );
+
+    public static final ObjectParser<GeoHexGridAggregationBuilder, String> PARSER = createParser(
+        NAME,
+        GeoHexGridAggregationBuilder::parsePrecision,
+        GeoHexGridAggregationBuilder::new
+    );
+
+    private static int parsePrecision(XContentParser parser) throws IOException, OpenSearchParseException {
+        final var token = parser.currentToken();
```

Comment:
can we be consistent in the using var or concrete classes in the repo. I can see the usage of both, lets try to stick to one.

Reply:
My aim is to use var in new PR. I don't have plan to refactor complete repo ( at this moment at least). At the same time, don't want to be constrained in not using new java features for sake of consistency. What do you think?

Reply:
I am not saying refractor the whole repo. But for the sake of this PR we should use 1 thing.

I feel we should consider consistency in the repo, rather than just bombarding the repo with different features. It's just make easier for code commiters to follow 1 specific pattern. You can think of it like the code formatter and linting we do in the repo.

I am nowhere saying we should use the new features, but lets use 1 thing as var is not providing any benefit on top of older one, I feel it reduces the code readability.

Reply:
updated this PR only with var.